R/modules.R

```diff
@@ -75,7 +75,7 @@ modules <- function(..., label = "root") {
     )
   }
 
-  checkmate::assert_list(submodules, min.len = 1, any.missing = FALSE, types = c("teal_module", "teal_modules"))
+  checkmate::assert_list(submodules, min.len = 1, any.missing = FALSE, types = c("teal_module", "teal_module_reporter", "teal_module_landing", "teal_modules"))
```

Comment:
Modules have a base extension `"teal_module"`. 
```suggestion
  checkmate::assert_list(submodules, min.len = 1, any.missing = FALSE, types = c("teal_module", "teal_modules"))
```